docs/platforms/javascript/guides/react/features/react-router.mdx

```diff
@@ -82,6 +82,55 @@ You can instrument [`createMemoryRouter`](https://reactrouter.com/en/main/router
 
 </Alert>
 
+### Custom Error Boundaries
+
+When using `react-router`, errors thrown inside route elements are only re-thrown in **development mode**. In production, these errors won't be surfaced unless manually captured. If you **don't** have a custom error boundary in place, `react-router` will create a default one that "swallows" all errors.
+
+<Alert>
+Note that this only applies to render method and lifecycle errors, since React doesn't need error boundaries to handle errors in event handlers.
+</Alert>
+
+To send errors to Sentry when using a custom error boundary, use the `Sentry.captureException` method:
+
+```jsx {11, 27}
+// router setup
+const sentryCreateBrowserRouter = wrapCreateBrowserRouter(createBrowserRouter);
+const router = sentryCreateBrowserRouter([
+  {
+    path: "/",
+    element: <YourLayout />,
+    children: [
+      {
+        path: "",
+        element: <Outlet />,
+        errorElement: <YourCustomRootErrorBoundary />,
+        children: [
+          // other routes ... 
+        ],
+      },
+    ],
+  },
+]);
+
+// error boundary
+import { useRouteError } from "react-router-dom";
+import * as Sentry from "@sentry/react";
+
+export function YourCustomRootErrorBoundary() {
+  const error = useRouteError() as Error;
+
+  Sentry.captureException(error);
```

Comment:
you might need to wrap this in a `useEffect` because it's a side effect - cache to the error